pkg/operator/oidcdiscoveryendpoint/controller.go

```diff
@@ -0,0 +1,420 @@
+	if err != nil {
+		return err
+	}
+
+	configclientset, err := configset.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	controllerRef := &corev1.ObjectReference{
+		Kind:      "deployment",
+		Namespace: operatorNamespace,
+		Name:      deploymentName,
+	}
+	eventRecorder := events.NewKubeRecorder(kubeclientset.CoreV1().Events(operatorNamespace), deploymentName, controllerRef)
+	logger := log.WithFields(log.Fields{"controller": controllerName})
+
+	if infraStatus.PlatformStatus == nil || infraStatus.PlatformStatus.AWS == nil {
+		return fmt.Errorf("Infrastructure platform status is not set")
+	}
+
+	r := &s3EndpointReconciler{
+		kubeclientset:      kubeclientset,
+		configclientset:    configclientset,
+		logger:             logger,
+		eventRecorder:      eventRecorder,
+		infrastructureName: infraStatus.InfrastructureName,
+		region:             infraStatus.PlatformStatus.AWS.Region,
+	}
+
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	p := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isServiceAccountTokenSigner(e.MetaNew)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isServiceAccountTokenSigner(e.Meta)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			// Deleting the token signer configmap shouldn't be done.
+			// Be safe and don't react.  When the openshift-kube-apiserver-operator
+			// recreates it, we'll catch it.
+			return false
+		},
+	}
+
+	// Create a namespace local cache separate from the Manager cache
+	// A namespace scoped cache can still handle cluster scoped resources
+	cache, err := cache.New(config, cache.Options{Namespace: tokenSignerConfigMapNamespace})
+	if err != nil {
+		return err
+	}
+
+	informer, err := cache.GetInformer(context.TODO(), &corev1.ConfigMap{})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Informer{Informer: informer}, &handler.EnqueueRequestForObject{}, p)
+	if err != nil {
+		return err
+	}
+
+	mgr.Add(&oidcDiscoveryEndpointController{reconciler: r, cache: cache, logger: logger})
+
+	return nil
+}
+
+// openshift-kube-apiserver/bound-sa-token-signing-certs[service-account-xxx.pub]
+func isServiceAccountTokenSigner(meta metav1.Object) bool {
+	// all managed resources are named pod-identity-webhook
+	if meta.GetName() == tokenSignerConfigMapName {
+		return true
+	}
+	return false
+}
+
+type s3EndpointReconciler struct {
+	kubeclientset      *kubernetes.Clientset
+	configclientset    *configset.Clientset
+	logger             log.FieldLogger
+	eventRecorder      events.Recorder
+	infrastructureName string
+	region             string
+}
+
+var _ reconcile.Reconciler = &s3EndpointReconciler{}
+
+func (r *s3EndpointReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.logger.Info("reconciling AWS S3 OIDC discovery endpoint on detected change to service account token signer")
```

Comment:
It's not just a change that triggers a Reconcile(), right (kube will periodically cause this to run even w/o a change)? Is it more accurate to remove the "detected change" text?